spec/app_api_spec.rb

```diff
@@ -0,0 +1,66 @@
+require "spec_helper"
+
+RSpec.describe "app API" do
+  context "when hitting a single room API point" do
+    before do # Overriding this constant so the response won't stay cached between examples
+      stub_const("CACHE_EXPIRY_TIMEOUT", 0)
+    end
+
+    it "returns a valid response when there are no events" do
+      stub_gcal_request(gcal_fake_response([]))
+      response = get "/room/zoom_a.json"
+      body = {"colour" => [255, 255, 255],
+              "enable_presence_device" => true,
+              "empty" => true,
+              "minutes_to_next_event" => false,
+              "minutes_to_end_of_event" => false,
+              "upcoming_event_today" => false,
+              "events" => []}
+
+      expect(JSON.parse(response.body)).to eq(body)
```

Comment:
This is a bit like testing that an error page on the website contains every single bit of HTML, instead of just the bit you're interested in. What I'd do is something like

```suggestion
      expect(JSON.parse(response.body.empty)).to be true
      expect(JSON.parse(response.body.events)).to be_empty
```

that way you're testing the two bits of information in the response you actually care about (does it say it's empty, and does the list of events agree?) without your test starting to fail if someone adds a new item to the response.

Reply:
I like that idea. Your suggestion is a much more robust test. I'll make the changes :-)